app/models/reaction.rb

```diff
@@ -30,6 +31,8 @@ class Reaction < ApplicationRecord
   before_save :assign_points
   after_create_commit :record_field_test_event
   after_commit :async_bust, :bust_reactable_cache, :update_reactable
+  after_commit :index_to_elasticsearch, if: proc { |r| r.category == "readinglist" && r.reactable && r.reactable.published }, on: %i[create update]
+  after_commit :remove_from_elasticsearch, if: proc { |r| r.category == "readinglist" && r.reactable && r.reactable.published }, on: [:destroy]
```

Comment:
these lines are quite long and they both use the same `if` callback, maybe they can share it if we transform it in a regular function